### PR TITLE
[POR-1794] Smart opt v2

### PR DESCRIPTION
**Pull request by sdess09**

### CODE REVIEW

File: dashboard/src/lib/hooks/useClusterResourceLimits.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }
+    const instanceType = data.labels["beta.kubernetes.io/instance-type"];
+    if (!instanceType) {
+        return defaultResources;
+    }
+    const splits = instanceType.split(".");
+    if (splits.length !== 2) {
+        return defaultResources;
+    }
+    const [instanceClass, instanceSize] = splits;
```
Comment (ianedwards) [inline dashboard/src/lib/hooks/useClusterResourceLimits.ts:28]: option to do something like:

`const [instanceClass, instanceSize] = z.tuple([z.string(), z.string()]).parse(instanceType)`

File: dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    max,
+    value,
+    setValue,
+    disabled,
+    disabledTooltip,
+    color,
+    step,
+    width,
+    isSmartOptimizationOn,
+    decimalsToRoundTo = 0,
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx:52]: just curious, will this prop ever be set / any reason for the default to not just be 2?

File: dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                value: min,
+                label: min.toString(),
+            }
+        ];
+
+        if (isSmartOptimizationOn) {
+            let half = min + (max - min) * 0.5;
+            let quarter = min + (max - min) * 0.25;
+            let eighth = min + (max - min) * 0.125;
+            if (decimalsToRoundTo > 0) {
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx:72]: potentially round to ints if 0?

**Review by ianedwards — APPROVED** [review]
nice

File: dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx (hunk 3 above)
Comment (ferozemohideen) [reply dashboard/src/main/home/app-dashboard/validate-apply/services-settings/tabs/IntelligentSlider.tsx]: now I just pass in 0 to .toFixed, which I think rounds to ints

**Pull request merged**